core/ide/src/main/java/org/alice/ide/ProjectApplication.java

```diff
@@ -388,6 +487,83 @@ private void projectLoaded(UserActivity activity, Project project) {
     hideWaitCursor();
   }
 
+  private boolean createProjectFromBackup(File backup, File original) {
+    YesNoCancelResult result = Dialogs.confirmOrCancel("Replace Project With Backup?",
+            "A backup has been opened successfully: " + backup.getName() + ".\n" +
```

Comment:
This needs to be tweaked to pass style check. Move +s to next line.
Also, all these strings should be internationalized, but that would be asking you to do as I say, not as I do.